hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -1412,6 +1413,7 @@ protected <R> Map<K, R> prepareResult(Collection<Entry<Data, Data>> entries) {
     @Override
     public <R> Map<K, R> executeOnEntries(@Nonnull EntryProcessor<K, V, R> entryProcessor,
                                           @Nonnull Predicate<K, V> predicate) {
+        PredicateUtils.checkDoesNotContainPagingPredicate(predicate, "executeOnEntries");
```

Comment:
minor: despite marking predicates as non nullable, we performed null checks in other APIs. Should we do the same here?